core/table.ts

```diff
@@ -135,8 +141,11 @@ export class Table {
   public dependencies(value: string | string[]) {
     const newDependencies = typeof value === "string" ? [value] : value;
     newDependencies.forEach(d => {
-      const table = this.session.tables[d];
-
+      const [fQd, err] = utils.matchFQName(d, this.session.getAllFQNames());
```

Comment:
this code is made complicated because dependency actions may not yet have been declared (when `dependencies()` is called). i think that we should collect all dependencies in `dependencies()` and then finally resolve them all in `compile()` (by which time all actions should have been declared).

Reply:
OK, will do. I see a pattern here where all action types should throw compilation errors only when `action.compile()` is run, and it makes sense to me.

A bit off-topic but for consistency reasons _perhaps_ the `dependencies` method should allow `Resolvable`'s or (arrays of) and become:

`public dependencies(ref: Resolvable | Resolvable[])`

Same with `preOps`, `postOps`, etc.